core/src/test/scala/io/finch/FinchSpec.scala

```diff
@@ -63,13 +52,16 @@ trait FinchSpec extends AnyFlatSpec with Matchers with Checkers with AllInstance
   }
 
   def genError: Gen[Error] = for {
-    i <- genRequestItem
     s <- genNonEmptyString
     e <- Gen.oneOf(
-      Error.NotPresent(i),
-      Error.NotParsed(i, implicitly[ClassTag[Int]], new Exception(s)),
-      Error.NotParsed(i, implicitly[ClassTag[err1.err2.Foo]], new Exception(s)),
-      Error.NotValid(i, s)
+      Error.BodyNotPresent,
+      Error.ParamNotPresent(s),
+      Error.HeaderNotPresent(s),
+      Error.CookieNotPresent(s),
+      Error.BodyNotParsed(implicitly[ClassTag[Int]]),
+      Error.ParamNotParsed(s, implicitly[ClassTag[Int]]),
+      Error.HeaderNotParsed(s, implicitly[ClassTag[err1.err2.Foo]]),
+      Error.CookieNotParsed(s, implicitly[ClassTag[err1.err2.Foo]])
```

Comment:
The idiomatic way is to use `classTag` from the `reflect` package to summon a `ClassTag`